src/app/profile/payment-methods/credit-card/credit-card.component.js

```diff
@@ -0,0 +1,84 @@
+import angular from 'angular';
+import template from './credit-card.tpl';
+import displayAddressComponent from 'common/components/display-address/display-address.component';
+import recurringGiftsComponent from '../recurring-gifts/recurring-gifts.component';
+import formatAddressForTemplate from 'common/services/addressHelpers/formatAddressForTemplate';
+import editPaymentMethodModal from 'common/components/paymentMethods/editPaymentMethod/editPaymentMethod.modal.component';
+import deletePaymentMethodModal from 'common/components/paymentMethods/deletePaymentMethod/deletePaymentMethod.modal.component.js';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+
+class CreditCardController{
+
+  /* @ngInject */
+  constructor(envService,$uibModal){
+    this.isCollapsed = true;
+    this.$uibModal = $uibModal;
+    this.imgDomain = envService.read('imgDomain');
+    this.submissionError = {error: ''};
+  }
+
+  $onInit(){
+    let address = this.model.creditcard ? this.model.creditcard.address : false;
+    this.formattedAddress = address ? formatAddressForTemplate(address) : false;
```

Comment:
Can you make the profile service do this? Just to be consistent. You can probably just copy what is here https://github.com/CruGlobal/give-web/blob/master/src/common/services/api/profile.service.js#L80-L88 or turn that into a function that can be called by both profile service methods.